alternator/auth.cc

```diff
@@ -133,7 +133,8 @@ future<std::string> get_key_from_roles(service::storage_proxy& proxy, std::strin
     }
     auto selection = cql3::selection::selection::for_columns(schema, {salted_hash_col});
     auto partition_slice = query::partition_slice(std::move(bounds), {}, query::column_id_vector{salted_hash_col->id}, selection->get_query_options());
-    auto command = ::make_lw_shared<query::read_command>(schema->id(), schema->version(), partition_slice, proxy.get_max_result_size(partition_slice));
+    auto command = ::make_lw_shared<query::read_command>(schema->id(), schema->version(), partition_slice,
+            proxy.get_max_result_size(partition_slice), query::tombstone_limit(proxy.get_tombstone_limit()));
     auto cl = auth::password_authenticator::consistency_for_user(username);
```

Comment:
Ok, it's called on the coordinator, good. Also saves us from different nodes having different configurations.

It's bad to involve replica::database in a coordinator problem, but we can clean that up later.

Reply:
We have the same problem with query max result size. The dilemma is that of where to put the logic of classifying the query, which is needed for both. After some back-and-forth `database` was selected, though it is by no means the perfect choice.